spec/racker_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Racker do
+  let(:app) { Rack::Builder.parse_file('config.ru').first }
+  let(:game) { Codebreaker::Entities::Game.new }
+  let(:path) { 'database/data_test.yml' }
+
+  before do
+    File.new(path, 'w+')
+    stub_const('Codebreaker::Entities::DataStorage::FILE_NAME', 'database/data_test.yml')
+    File.write(Codebreaker::Entities::DataStorage::FILE_NAME, [].to_yaml)
+  end
+
+  after { File.delete(path) }
+
+
+  describe 'statuses' do
+    context 'when root path' do
+      before { get '/' }
+
+      it 'returns status ok' do
+        expect(last_response).to be_ok
+      end
+
+      it { expect(last_response.body).to include I18n.t(:short_rules) }
+    end
+
+    context 'when unknown routes' do
+      before { get '/unknown' }
+
+      it 'returns status not found' do
+        expect(last_response.body).to include I18n.t(:not_found)
+      end
+    end
+
+    context 'when rules path' do
+      before { get '/rules' }
+
+      it 'returns status ok' do
+        expect(last_response).to be_ok
+      end
+
+      it { expect(last_response.body).to include I18n.t(:cb_rules) }
+    end
+
+    context 'when statistics path' do
+      before do
+        env 'rack.session', scores: []
+        get '/stats'
+      end
+
+      it { expect(last_response.body).to include I18n.t(:top_of_players) }
+      it { expect(last_response).to be_ok }
+    end
+  end
+
+  describe '#hint' do
+    before do
+      game.generate(Codebreaker::Entities::Game::DIFFICULTIES[:hell])
+      env 'rack.session', game: game, used_hints: [], level: :easy
+      get '/hint'
+    end
+
+    it 'adds value to session hints array' do
+      post '/hint'
+      expect(last_request.session[:used_hints]).not_to be_empty
+      expect(last_request.session[:game].code.join.include?(last_request.session[:used_hints].join)).to be true
```

Comment:
https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/built-in-matchers/include-matcher